indico/modules/events/papers/models/papers.py

```diff
@@ -66,3 +68,45 @@ def state(self):
     @state.setter
     def state(self, state):
         self.last_revision.state = state
+
+    def is_in_final_state(self):
+        return self.state in {PaperRevisionState.accepted or PaperRevisionState.rejected}
+
+    def can_comment(self, user):
+        if not user:
```

Comment:
This is starting to become a quite repeated pattern for protection checks. Maybe it would be interesting having a decorator `@protection_check` that does it for you. /cc @ThiefMaster